src/Microsoft.TemplateEngine.Abstractions/IParameterSet.cs

```diff
@@ -1,31 +1,297 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
+using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.Logging;
+
+#pragma warning disable RS0016 // Add public types and members to the declared API
+#pragma warning disable SA1507 // Code should not contain multiple blank lines in a row
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable SA1201 // Elements should appear in the correct order
+#pragma warning disable SA1516 // Elements should be separated by blank line
+#pragma warning disable SA1202 // Elements should be ordered by access
 
 namespace Microsoft.TemplateEngine.Abstractions
 {
+    public interface IParameterDefinitionsSet : IEnumerable<ITemplateParameter>, IReadOnlyDictionary<string, ITemplateParameter>
+    {
+        /// <summary>
+        /// Casts the Parameter set to unambiguous enumerable.
+        /// </summary>
+        /// <returns></returns>
+        IEnumerable<ITemplateParameter> AsEnumerable();
+    }
+
     /// <summary>
     /// Defines a set of template parameters.
     /// </summary>
-    public interface IParameterSet
+    public interface IParameterSet2
```

Comment:
should we keep `IParameterSet` then?

Reply:
Yes - I'll rename. It was for in-progress changes wehere I was moving to the new type